### PR TITLE
init bicep

### PR DESCRIPTION
**Pull request by jongio**

### CODE REVIEW

**Comment by jongio** [comment]
@alex-frankel

File: iac/bicep/role.bicep [hunk 1]
```diff
@@ -0,0 +1,12 @@
+param roleId string = ''
+param principalType string = 'ServicePrincipal'
+param principalId string = ''
+
+resource role_assignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(subscription().id, principalId, roleId)
```
Comment (alex-frankel) [inline iac/bicep/role.bicep:6]: The `guid()` will take n arguments, so we can keep making the seed more complex. I wonder if we can make this work by adding more arguments to the `guid()` function like a resource ID or the resourceGroup ID if applicable.

Comment (jongio) [reply iac/bicep/role.bicep]: Added rgName

**Comment by jongio** [comment]
@anthony-c-martin

File: iac/bicep/deploy.sh [hunk 2]
```diff
@@ -5,21 +5,4 @@ export ROOT=../..;source $ROOT/scripts/base.sh
...
-
-# Azure Function MI Permissions to KV
-# We need this becuase of the cyclic dependency on KV and AF
-FUNCTION_MI_ID=$(az functionapp identity show -n memealyzerdevfunction -g memealyzerdevrg --query principalId -o tsv)
-az keyvault set-policy -g ${BASENAME}rg -n ${BASENAME}kv --object-id $FUNCTION_MI_ID --secret-permissions get set list delete
-
-# Apply all the necessary permissions that we couldn't get to work with bicep
-source ../scripts/perms.sh ${BASENAME}
\ No newline at end of file
+az deployment sub create --debug -n ${BASENAME} -l ${REGION} -f main.json -p basename=${BASENAME} location=${REGION} failover_location=${FAILOVER_REGION} cli_user_id=${CLI_USER_ID}
```
Comment (alex-frankel) [inline iac/bicep/deploy.sh:8]: one-liner!

**Comment by anthony-c-martin** [comment]
Looks great! Personally I think I would gitignore the generated json files in the `iac/bicep` directory as they should be generated on the, but that's obviously just a preference :)

**Review by alex-frankel — APPROVED** [review]
Looks awesome - glad to see we were able to get everything working inside of bicep like permissions and queues

**Pull request merged**